scrape/manager.go

```diff
@@ -223,10 +222,24 @@ func (m *Manager) TargetsActive() map[string][]*Target {
 	m.mtxScrape.Lock()
 	defer m.mtxScrape.Unlock()
 
+	var (
+		wg  sync.WaitGroup
+		mtx sync.Mutex
+	)
+
 	targets := make(map[string][]*Target, len(m.scrapePools))
+	wg.Add(len(m.scrapePools))
 	for tset, sp := range m.scrapePools {
-		targets[tset] = sp.ActiveTargets()
+		// Running in parallel limits the blocking time of scrapePool to scrape
+		// interval when there's an update from SD.
+		go func(tset string, sp *scrapePool) {
+			mtx.Lock()
```

Comment:
Does this really run in parallel?  As far as I can see it's all serialized by the mutex - what did you gain?